brave/brave-core
# Add four P3A metrics for Playlist

Resolves https://github.com/brave/brave-browser/issues/29913



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Enable playlist flag before testing.

For `Brave.Playlist.LastUsageTime`:

0. Start with fresh profile. Ensure metric does not exist.
1. Use Playlist for the first time (view the list of playlists or add an item to a playlist), ensure the metric has a value of 1.
2. Advance the system time by one week, restart app, do not use Playlist. Ensure metric has value of 2.
3. Test other values by repeating step 2 with different time intervals.
4. Ensure metric has value of 1 after using Playlist again.

For `Brave.Playlist.NewUserReturning`:

0. Start with fresh profile. Ensure metric does not exist.
1. Use Playlist for the first time (view the list of playlists or add an item to a playlist), ensure the metric has a value of 2.
2. Advance the system time by one day, restart app, do not use Playlist. Ensure metric still has value of 2.
3. Use playlist, ensure metric has value of 3.
4. Advance system time by one week, restart app. Ensure metric has value of 1. Use playlist and ensure the metric value is the same.

For `Brave.Playlist.UsageDaysInWeek`:

0. Start with fresh profile. Ensure metric does not exist.
1. Use Playlist for the first time (view the list of playlists or add an item to a playlist), ensure the metric has a value of 1.
2. Advance the system time by one day, restart app, use Playlist. Repeat this step one more time. Ensure metric has value of 2.
3. Test other values by repeating step 2 with different time intervals.
4. Wait until metric is sent. Once sent, advance system time by one week and restart app. The metric should no longer exist.
5. Use playlist again. The metric should reappear with the correct value.

For `Brave.Playlist.FirstTimeOffset`:

0. Start with fresh profile. Ensure metric does not exist.
1. Use Playlist for the first time (view the list of playlists or add an item to a playlist), ensure the metric has a value of 1.
2. Advance the system time by one week, restart app, use Playlist. Ensure the metric has the same value.
3. Wait until metric is sent. Once sent, advance system time by one week and restart app. The metric should no longer exist.
4. Use Playlist again, ensure metric still does not exist.
5. Repeat this test plan and advance system time by different intervals before step 1 to test different metric values.
